website/www/site/content/en/documentation/sdks/python-custom-multi-language-pipelines-guide.md

```diff
@@ -0,0 +1,307 @@
+        beam:schematransform:org.apache.beam:my_transform:v1
+```
+
+The transform is discoverable at `localhost:12345`. Foreign SDKs can now discover and add it to their pipelines. The next section demonstrates how to do this with a Python pipeline.
+
+## Use the portable transform in a Python pipeline
+
+The Python SDK’s [**ExternalTransformProvider**](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.external_transform_provider.html#apache_beam.transforms.external_transform_provider.ExternalTransformProvider)
+can dynamically generate wrappers for portable transforms.
+
+```python
+from apache_beam.transforms.external_transform_provider import ExternalTransformProvider
+```
+
+### Connect to an expansion service
+First, connect to an expansion service that contains the transform. This section demonstrates two methods of connecting to the expansion service.
+
+#### Connect to an already running service
+
+If your expansion service JAR file is already running, pass in the address:
+
+```python
+provider = ExternalTransformProvider("localhost:12345")
+```
+
+#### Start a service based on a Java JAR file
+
+If the service lives in a JAR file but isn’t currently running, use Beam utilities to run the service in a subprocess:
+
+```python
+from apache_beam.transforms.external import JavaJarExpansionService
+
+provider = ExternalTransformProvider(
+    JavaJarExpansionService("path/to/my-expansion-service.jar"))
+```
+
+You can also provide a list of services:
+
+```python
+provider = ExternalTransformProvider([
+    "localhost:12345",
+    JavaJarExpansionService("path/to/my-expansion-service.jar"),
+    JavaJarExpansionService("path/to/another-expansion-service.jar")])
+```
+
+When initialized, the `ExternalTransformProvider` connects to the expansion service(s), retrieves all portable transforms, and generates a Pythonic wrapper for each one.
+
+### Retrieve and use the transform
+
+Retrieve the transform using its unique identifier and use it in your multi-language pipeline:
+
+```python
+identifier = "beam:schematransform:org.apache.beam:my_transform:v1"
+MyTransform = provider.get_urn(identifier)
+
+with beam.Pipeline() as p:
+  p | beam.Create(...) | MyTransform(foo="abc", bar=123)
+```
+
+
+### Inspect the transform's metadata
+You can learn more about a portable transform’s configuration by inspecting its metadata:
+
+```python
+import inspect
+
+inspect.getdoc(MyTransform)
+# Output: "This transform does this and that..."
+
+inspect.signature(MyTransform)
+# Output: (foo: "str: Description of what foo does...",
+#	     bar: "int: Description of what bar does....")
+```
+
+This metadata is generated directly from the provider's implementation. The class documentation is generated from the [optional **description** method](#additional-metadata). The signature information is generated from the `@SchemaFieldDescription` annotations in the [configuration object](#implement-a-configuration).
+
```

Comment:
do we have a full example somewhere for this guide?

Reply:
I'm adding a quickstart that uses an example with the same format here #33360

Reply:
make sure we link it here.